docs/deploy/nginx-uwsgi.rst

```diff
@@ -0,0 +1,246 @@
+  $ /home/myproject/venv/bin/pip install uwsgi
+
+
+.. note::
+
+  The exact commands for creating a virtual environment might differ based on
+  the Python version you are using and the system you are on. At the end of the day
+  we need a virtualenv in /home/myproject/venv with your project dependencies
+  installed. Use the ``pip`` binary within the virtual environment by ``source
+  venv/bin/activate`` or using the full path.
+
+
+Okay, that is all for getting your application on the server and setting up the
+environment.
+
+Preparing your Application for Service
+''''''''''''''''''''''''''''''''''''''
+
+You likely have a way to configure your application, such as with a
+``create_api()`` function or a module level script. This function or script
+needs to supply an instance of a :any:`falcon.API`.
+
+We need to expose that instance of the :any:`falcon.API` in some way so that
+uWSGI can find it; we will do that by creating a ``wsgi.py`` file. Modify the
+content of this file based on how your application is configured.  Ensure that
+you expose  a variable called ``application`` which is assigned to your
+:any:`falcon.API` instance.
+
+.. code-block:: python
+  :caption: /home/myproject/src/wsgi.py
+
+  import os
+  import myproject
+
+  config = myproject.get_config(os.environ['MYPROJECT_CONFIG'])
+  application = myproject.create_api(config)
+
+The above example shows how to implement a bare-bones ``wsgi.py``. It imports your
+application and creates the :any:`falcon.API` instance that represents our
+application, exposing it as the module-scoped ``application`` variable. uWSGI
+will use this file to load your application.
+
+Note that we did not call ``run`` like a developer might do during development
+or use `wsgiref.simple_server.make_server`.  Instead we simply assigned our
+`API` instance to ``application``. We *do not* want to start a server, since
+uWSGI does that automatically. Starting an independent WSGI server in your
+``wsgi.py`` file will render unexpected results.
+
+
+Deploying Falcon behind uWSGI
+'''''''''''''''''''''''''''
+
+With our ``wsgi.py`` file in place, it is time to configure uWSGI.  To do this,
+we create a ``uwsgi.ini`` file. In general, you shouldn’t commit this file to
+source control; it should be generated from a template by your deployment
+toolchain according to the target environment (number of CPUs, etc.).
+
+This configuration, when executed, will create a new uWSGI server pointed at our
+wsgi.py file and listening at ``12.0.0.1:8080``.
+
+.. code-block:: ini
+  :caption: /home/myproject/src/uwsgi.ini
+
+  [uwsgi]
+  master = 1
+  vacuum = true
+  socket = 127.0.0.1:8080
+  enable-threads = true
+  thunder-lock = true
+  threads = 2
+  processes = 2
+  virtualenv = /home/myproject/venv
+  wsgi-file = /home/myproject/src/wsgi.py
+  chdir = /home/myproject/src
+  uid = myproject-runner
+  gid = myproject-runner
+
+
+.. note:: Thread vs Processes vs gevent
```

Comment:
I know I have already been nagging about this, but mentioning *gevent* only here in the title and nowhere else feels a bit Spartan. Could we consider throwing in a sentence or two, ideally with a uWSGI configuration example?

Reply:
I removed gevent for now. I don't have the time to explore that and I am not familiar with running gevent. I am certainly open to suggestions if you have it.